osu.Game/Screens/Select/BeatmapDetails.cs

```diff
@@ -0,0 +1,525 @@
+
+            private float difficultyValue;
+
+            public float Value
+            {
+                get
+                {
+                    return difficultyValue;
+                }
+                set
+                {
+                    difficultyValue = value;
+                    bar.Length = value / maxValue;
+                    valueText.Text = value.ToString(CultureInfo.InvariantCulture);
+                }
+            }
+
+            private float maxValue = 10;
+
+            public float MaxValue
+            {
+                get
+                {
+                    return maxValue;
+                }
+                set
+                {
+                    maxValue = value;
+                    bar.Length = Value / value;
+                }
+            }
+
+            public string DifficultyName
+            {
+                get
+                {
+                    return name.Text;
+                }
+                set
+                {
+                    name.Text = value;
+                }
+            }
+
+            public Color4 AccentColour
+            {
+                get
+                {
+                    return bar.AccentColour;
+                }
+                set
+                {
+                    bar.AccentColour = value;
+                }
+            }
+
+            public DifficultyRow()
+            {
+                Children = new Drawable[]
+                {
+                    name = new OsuSpriteText
+                    {
+                        Font = @"Exo2.0-Regular",
+                    },
+                    bar = new Bar
+                    {
+                        Origin = Anchor.CentreLeft,
+                        Anchor = Anchor.CentreLeft,
+                        RelativeSizeAxes = Axes.Both,
+                        Size = new Vector2(1, 0.35f),
+                        Padding = new MarginPadding { Left = 100, Right = 25 },
+                    },
+                    valueText = new OsuSpriteText
+                    {
+                        Anchor = Anchor.TopRight,
+                        Origin = Anchor.TopRight,
+                        Font = @"Exo2.0-Regular",
+                    },
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colour)
+            {
+                name.Colour = colour.GrayB;
+                bar.BackgroundColour = colour.Gray7;
+                valueText.Colour = colour.GrayB;
+            }
+        }
+
+        private class MetadataSegment : Container, IHasAccentColour
+        {
+            private readonly OsuSpriteText header;
+            private readonly FillFlowContainer<OsuSpriteText> content;
+
+            private const int fade_time = 250;
+
+            public string HeaderText
+            {
+                set
+                {
+                    header.Text = value;

```

Comment:
this never changes right? and never should change.

just make it a constructor argument maybe?